controllers/platformnetwork_controller.go

```diff
@@ -1325,3 +1334,56 @@ func (r *PlatformNetworkReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&starlingxv1.PlatformNetwork{}).
 		Complete(r)
 }
+
+// Verify whether we have annotation restore-in-progress
+func (r *PlatformNetworkReconciler) checkRestoreInProgress(instance *starlingxv1.PlatformNetwork) bool {
+	restoreInProgress, ok := instance.Annotations[cloudManager.RestoreInProgress]
+	if ok && restoreInProgress != "" {
+		return true
+	}
+	return false
+}
+
+// Update status and clear annoatation
+func (r *PlatformNetworkReconciler) RestorePlatformNetworkStatus(instance *starlingxv1.PlatformNetwork) error {
```

Comment:
Please avoid creating nested functions. I recommend creating additional helper functions instead.

Reply:
Additional helper function for clearing annotation is created